src/content.ts

```diff
@@ -1135,6 +1136,19 @@ declare global {
   function onPageHide() {
     browser.runtime.sendMessage({ type: 'disabled' });
   }
+
+  window.addEventListener(
+    'DOMContentLoaded',
+    (event) => {
+      const rp = new RikaiPuck();
+      rp.render(document.body);
+      rp.enable();
+    },
+    {
+      once: true,
+    },
+  );
```

Comment:
We could probably do this without waiting on `DOMContentLoaded`. The reason being that the content script can be injected after `DOMContentLoaded` is fired (in fact, for the activeTab case it almost always is).

Also we should render the puck as part of the shadow tree used for the pop-up (or a separate shadow tree). This will give us proper style isolation from the host page.

I'm happy to make up patches for these changes once we decide how to proceed on this work.

Reply:
I'd be grateful for you to take on those two tasks! I guess for the `DOMContentLoaded` one, we'd do an additional check for whether the document has already been loaded? Either way, I'll leave that to you.

Happy for you to turn it into a shadow tree to see how it's done – new territory for myself.

Reply:
Thanks! I just tweaked this over in https://github.com/shirakaba/10ten-ja-reader/pull/1. (I had to check for `readyState !== 'loading'` in order to cover the `readyState === 'complete'` case.)